src/ape/managers/compilers.py

```diff
@@ -179,6 +181,37 @@ def compile(self, contract_filepaths: List[Path]) -> Dict[str, ContractType]:
 
         return contract_types_dict
 
+    def compile_source(self, compiler_name: str, code: str, **kwargs) -> ContractContainer:
```

Comment:
Now I am thinking about this in combo with https://github.com/ApeWorX/ape/pull/1705 and wondering if we should only have a compiler settings optional argument here